frontend/src/components/Header/Header.tsx

```diff
@@ -1,24 +1,37 @@
 import AppBar from 'material-ui/AppBar'
+import FlatButton from 'material-ui/FlatButton'
 import IconButton from 'material-ui/IconButton'
 import GithubIcon from 'mui-icons/cmdi/github'
 import * as React from 'react'
+import { withStore } from '../../services/store'
+import { AboutDialog } from '../AboutDialog/AboutDialog'
 import './Header.css'
 
 let title = 'Encompass' + (process.env.TITLE_SUFFIX ? ' - ' + process.env.TITLE_SUFFIX : '')
 
-export let Header: React.StatelessComponent = () =>
-  <AppBar
+export let Header = withStore('isAboutDialogOpen')(({ store }) =>
```

Comment:
why does it have to go into the store?

Reply:
easier to handle open  / close. Other suggestions? Just felt weird to use a state when we use it nowhere else

Reply:
yeah idk we're not using it outside of this component so seems strange to keep it in the store. maybe @bcherny can help us decide

Reply:
Agreed that the `isOpen` should just be part of this component's `state`. Doesn't need to be in the store.

Reply:
I like putting everything in the store, but it's mostly a stylistic choice. A few benefits of putting it in the store:

- Easy to add effects to do stuff when users open the modal in the future (eg. close other open modals)
- Easy to audit app-wide state changes by using `withLogger` and opening devtools
- Easy mental model: all state lives in 1 place
- 1 line of code to add an effect to track this state (or any other state) with Google Analytics in the future
- Ability to easily snapshot the whole app state for testing, bug reports, etc. in the future

A downside:

- State is non-local, and takes a CTRL+F to find where a piece of state in the store is used

Reply:
Thanks for the recap! Super helpful :)